ostd/src/arch/x86/cpu/local.rs

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Architecture dependent CPU-local information utilities.
+
+use x86_64::registers::segmentation::{Segment64, FS};
+
+/// Sets the base address for the CPU local storage by writing to the FS base model-specific register.
+/// This operation is marked as `unsafe` because it directly interfaces with low-level CPU registers.
+///
+/// # Safety
+///
+///  - This function is safe to call provided that the FS register is dedicated entirely for CPU local storage
+///    and is not concurrently accessed for other purposes.
+///  - The caller must ensure that `addr` is a valid address and properly aligned, as required by the CPU.
+///  - This function should only be called in contexts where the CPU is in a state to accept such changes,
+///    such as during processor initialization.
+pub(crate) unsafe fn set_base(addr: u64) {
+    FS::write_base(x86_64::addr::VirtAddr::new(addr));
+}
+
+/// Gets the base address for the CPU local storage by reading the FS base model-specific register.
+pub(crate) fn get_base() -> u64 {
+    FS::read_base().as_u64()
+}
+
+pub mod preempt_lock_count {
+    //! We need to atomically increment and decrement the per-CPU preemption
+    //! lock count, requiring the implementation using a single instruction.
```

Comment:
Don't use the word "atomic", just say that we need to "use a single instruction".

`inc` is not an atomic operation as it does not synchronize between CPU cores. For an atomic operation, we need to use `lock; inc`. But using `lock; inc` is not necessary here since the preemption count is per-CPU.

This also applies to all the comments below, like:
```patch
-   /// Atomically increments the per-CPU preemption lock count.
+   /// Increments the per-CPU preemption lock count using a single instruction.
```